scaper/core.py

```diff
@@ -358,6 +359,84 @@ def _validate_distribution(dist_tuple):
                 'number that is equal to or greater than trunc_min.')
 
 
+def _ensure_satisfiable_source_time_tuple(source_time, source_duration, event_duration):
+    '''
+    Modify a source_time distribution tuple according to the duration of the
+    source and the duration of the event. This allows you to sample from 
+    anywhere in a source file without knowing the exact duration of every
+    source file. 
+
+    Parameters
+    ----------
+    source_time : tuple
+        Tuple specifying a distribution to sample from. See Scaper.add_event
+        for details about the expected format of the tuple and allowed values.
+    source_duration : float
+        Duration of the source audio file.
+    event_duration : float
+        Duration of the event to be extracted from the source file.
+
+    See Also
+    --------
+    Scaper.add_event : Add a foreground sound event to the foreground
+    specification.
+    '''
+    _validate_distribution(source_time)
+    old_source_time = deepcopy(source_time)
+    source_time = list(source_time)
+
+    # If it's a constant distribution, just make sure it's within bounds.
+    if source_time[0] == 'const':
+        if source_time[1] + event_duration > source_duration:
+            source_time[1] = max(0, source_duration - event_duration)
+
+    # If it's a choose, iterate through the list to make sure it's all in bounds.
+    # Some logic here so we don't add stuff out of bounds more than once.
+    elif source_time[0] == 'choose':
+        for i, t in enumerate(source_time[1]):
+            if t + event_duration > source_duration:
+                source_time[1][i] = max(0, source_duration - event_duration)
+
+    # If it's a uniform distribution, tuple must be of length 3, We change the 3rd
+    # item to source_duration - event_duration so that we stay in bounds. If the min
+    # out of bounds, we change it to be source_duration - event_duration - eps.
```

Comment:
I believe there's no use of eps anymore, we can update the comment

Reply:
Done.